bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/StylesheetInlinerTransformerFactory.java

```diff
@@ -0,0 +1,200 @@
+
+            String[] selectors = slingRequest.getRequestPathInfo().getSelectors();
+
+            if (selectors.length > 0) {
+                enabled = Arrays.asList(selectors).contains(SELECTOR);
+                if (enabled) {
+                    log.info("Inlining Stylesheet references for {}", slingRequest.getRequestURL().toString());
+                }
+            }
+        }
+
+        public void startElement(final String namespaceURI, final String localName, final String qName,
+                final Attributes attrs) throws SAXException {
+            if (enabled) {
+                try {
+                    if (isCSS(localName, attrs)) {
+                        String sheet = attrs.getValue("", "href");
+                        if (inHead) {
+                            stylesheetsInHead.add(sheet);
+                        } else {
+                            log.debug("Inlining stylesheet link found in BODY: '{}'", sheet);
+                            inlineSheet(namespaceURI, sheet);
+                        }
+                    } else {
+                        getContentHandler().startElement(namespaceURI, localName, qName, attrs);
+
+                        if (localName.equalsIgnoreCase(HEAD)) {
+                            inHead = true;
+                        } else if (localName.equalsIgnoreCase(BODY)) {
+
+                            // add each of the accumulated stylesheet references
+                            for (String sheet : stylesheetsInHead) {
+                                log.debug("Inlining sheet found in HEAD: '{}'", sheet);
+                                inlineSheet(namespaceURI, sheet);
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    log.error("Exception in stylesheet inliner", e);
+                    throw new SAXException(e);
+                }
+            } else {
+                getContentHandler().startElement(namespaceURI, localName, qName, attrs);
+            }
+        }
+
+        private void inlineSheet(final String namespaceURI, String s) throws Exception {
+            InputStream inputStream = null;
+
+            String withoutExtension = s.substring(0, s.indexOf(LibraryType.CSS.extension));
+            HtmlLibrary library = htmlLibraryManager.getLibrary(LibraryType.CSS, withoutExtension);
+            if (library != null) {
+                inputStream = library.getInputStream();
+            } else {
+                Resource resource = slingRequest.getResourceResolver().getResource(s);
+
+                if (resource != null) {
+                    inputStream = resource.adaptTo(InputStream.class);
+                }
+            }
+
+            if (inputStream != null) {
+                InputStreamReader isr = new InputStreamReader(inputStream);
+
+                CharArrayWriter caw = new CharArrayWriter(inputStream.available());
```

Comment:
I think it would be better to use` IOUtils.toCharArray()` here just to avoid the extra few lines of code. That would involve two calls to `characters()`. Or you could use `IOUtils.copy()` if you wanted to preserve the `CharArrayWriter`.